### ISSUE TITLE
Queries that reference other fields

### ISSUE THREAD

**Issue by a-type** [tantaman/aphrodite#37]
I know it's early on, but I'm just tinkering with this. One thing I'd like to express in a query is a predicate which is related to another field on a model.

Example use case: a grocery list app which lets you change the quantity of items. Suppose you've already purchased 6 eggs, but you increase the desired quantity to 12. In order to express a query for "purchased items," I want to query all items where `item.purchasedQuantity >= item.desiredQuantity`.

The scope here would only be predicates relevant to other fields on the same model, not traversing edges or anything like that.

**Comment by a-type** [comment]
Nevermind, I only just realized there's a plain `.where`!

**Comment by tantaman** [comment]
I should probably put big red warnings on the plain where. It doesn’t hoist
the provided lambda to the backend. In other words, the filtering for the
raw where is done in js rather than in sql. Fine for small amounts of data.

**Comment by tantaman** [comment]
Reopening since a hoisted (optimized) version of this would be good to have.

**Comment by a-type** [comment]
Yeah, makes sense. Still glad it's part of the API, since the alternative is doing the JS filtering myself, which won't have any chance opt me into that optimization when it's available.

**Comment by tantaman** [comment]
I added `whereAsync` and `mapAsync` -- https://github.com/tantaman/aphrodite/commit/ce7d5eb00d2fdadfe54b77e90136ae4204e69c1c

But still need to support `whereExists` e.g., https://www.sqlitetutorial.net/sqlite-exists/

**Comment by tantaman** [comment]
Will also add raw sql support to unblock people until such a point where everything can be directly supported.

E.g., `whereRaw(sql`your raw sql filter here`)`
